src/util/records.py

```diff
@@ -597,6 +597,17 @@ def rename_xmlid(cr, old, new, noupdate=None, on_collision="fail"):
                 # iif the key has been updated for this view, also update it for all other cowed views.
                 # Don't change the view keys inconditionally to avoid changing unrelated views.
                 cr.execute("UPDATE ir_ui_view SET key = %s WHERE key = %s", [new, old])
+
+        for inh in inherit_parents(cr, model, recursive=False):
+            if inh.via:
+                parent = inh.model.replace(".", "_")
+                rename_xmlid(
+                    cr,
+                    "{}_{}".format(old, parent),
+                    "{}_{}".format(new, parent),
+                    noupdate=noupdate,
+                    on_collision=on_collision,
+                )
```

Comment:
Do we need the `recursive=False` only for this?
```suggestion
        inh = next(inherit_parents(cr, model), None)
        if inh and inh.via:
            parent = inh.model.replace(".", "_")
            rename_xmlid(
                cr,
                "{}_{}".format(old, parent),
                "{}_{}".format(new, parent),
                noupdate=noupdate,
                on_collision=on_collision,
            )
```

Reply:
Actually from the error in runbot maybe `parent = next(inherit_parents(cr, model), None)` is enough?

Reply:
Nothing forbid models to have multiple `_inherits`.

Reply:
I see...